DDCore/src/plugins/VolumeMgrTest.cpp

```diff
@@ -81,99 +84,177 @@ VolIDTest::VolIDTest(LCDD& lcdd, DetElement sdet, size_t depth) : m_det(sdet) {
+          for (size_t i = chain.size(); i > 0; --i)  {
+            const TGeoMatrix* mat = chain[i-1]->GetMatrix();
+            ::printf("Placement [%d]  VolID:%s\t\t",int(i),chain[i-1].volIDs().str().c_str());
+            mat->Print();
+          }
+          ::printf("Computed Trafo (from placements):\t\t");
+          trafo.Print();
+          det_elem  = m_mgr.lookupDetElement(vid);
+          ::printf("DetElement Trafo: %s [%s]\t\t",
+                   det_elem.path().c_str(),volumeID(det_elem.volumeID()).c_str());
+          det_elem.nominal().worldTransformation().Print();
+          ::printf("VolumeMgr  Trafo: %s \t\t",det_elem.path().c_str());
+          m_mgr.worldTransformation(vid).Print();
+          int ii=1;
+          DetElement par = det_elem;
+          while( (par.isValid()) )  {
+            const TGeoMatrix* mat = par.placement()->GetMatrix();
+            ::printf("Element placement [%d]  VolID:%s %s\t\t",int(ii),
+                     par.placement().volIDs().str().c_str(), par.path().c_str());
+            mat->Print();
+            par = par.parent();
+            ++ii;
+          }
+        }
+        else  {
+        }
+      }
+    }
+    catch(const exception& ex) {
+      err << "Matrix " << pv.name() << " id:" << volumeID(vid)
+          << " path:" << detector.path() << " error:" << ex.what();
+    }
+    
   }
 }
 
 /// Walk through tree of detector elements
-void VolIDTest::walkVolume(DetElement e, PlacedVolume pv, VolIDs ids, size_t depth, size_t mx_depth)  const   {
+void VolIDTest::walkVolume(DetElement detector, PlacedVolume pv, VolIDs ids, const Chain& chain,
+                           size_t depth, size_t mx_depth)  const
+{
   if ( depth <= mx_depth )  {
-    const TGeoNode* current = pv.ptr();
-    TObjArray*  nodes  = current->GetNodes();
-    int   num_children = nodes ? nodes->GetEntriesFast() : 0;
+    const TGeoNode* current  = pv.ptr();
+    TObjArray*  nodes        = current->GetNodes();
+    int         num_children = nodes ? nodes->GetEntriesFast() : 0;
+
     for(int i=0; i<num_children; ++i)   {
       TGeoNode* node = (TGeoNode*)nodes->At(i);
       PlacedVolume place(node);
       VolIDs child_ids(ids);
+      Chain  child_chain(chain);
 
-      child_ids.insert(child_ids.end(),place.volIDs().begin(),place.volIDs().end());
-      bool is_sensitive = place.volume().isSensitive();
-      if ( is_sensitive )  {
-        checkVolume(e,place,child_ids);
-      }
-      walkVolume(e,place,child_ids,depth+1,mx_depth);
+      place.access(); // Test validity
+      child_chain.push_back(place);
+      child_ids.insert(child_ids.end(), place.volIDs().begin(), place.volIDs().end());
+      //bool is_sensitive = place.volume().isSensitive();
+      //if ( is_sensitive || !child_ids.empty() )  {
+        checkVolume(detector, place, child_ids, child_chain);
+        //}
+      walkVolume(detector, place, child_ids, child_chain, depth+1, mx_depth);
     }
   }
 }
-
+#if 0
```

Comment:
This test will be further elaborated in the future, but I do not yet know exactly what should go there. So let's keep it.